h2/src/main/org/h2/expression/Function.java

```diff
@@ -1494,6 +1495,49 @@ private Value getValueWithArgs(Session session, Expression[] args) {
         case DATE_DIFF:
             result = ValueLong.get(datediff(v0.getString(), v1, v2));
             break;
+        case DATE_TRUNC:   
+            // Retrieve the field value (e.g. 'day', 'microseconds', etc.).
+            String fieldValue = v0.getString().toLowerCase();
+
+            // Retrieve the dateValue.
```

Comment:
If you're planning to add more date-time units, it's better to extract all code below into a separate method.